File: API/Entities/LabBooking.cs
using System;

namespace API.Entities;

public class LabBooking
{
    public int Id { get; set; }

    public string UserName { get; set; } = string.Empty;

    public string WeekDays { get; set; } = string.Empty; // Monday–Saturday

    public TimeOnly StartTime { get; set; }

    public TimeOnly EndTime { get; set; }
    public string? Description { get; set; }


    public DateOnly BookingDate { get; set; }

}

File: client/src/app/_models/labbooking.ts
export interface LabBooking {
    id?: number;            // Optional during creation
    userName?: string;    // Optional when posting a new booking (filled on the server)
    firstName?: string;     //   Added to match backend DTO
    lastName?: string;      //   Added to match backend DTO
    weekDays: string;
    startTime: string;      // 'HH:mm' format
    endTime: string;        // 'HH:mm' format
    bookingDate: string;    // 'yyyy-MM-dd' format
    description?: string;
}
